elasticsearch/src/main/java/org/hibernate/search/elasticsearch/schema/impl/DefaultElasticsearchSchemaValidator.java

```diff
@@ -0,0 +1,235 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.elasticsearch.schema.impl;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import org.hibernate.search.elasticsearch.logging.impl.Log;
+import org.hibernate.search.elasticsearch.schema.impl.model.DataType;
+import org.hibernate.search.elasticsearch.schema.impl.model.DynamicType;
+import org.hibernate.search.elasticsearch.schema.impl.model.IndexMetadata;
+import org.hibernate.search.elasticsearch.schema.impl.model.IndexType;
+import org.hibernate.search.elasticsearch.schema.impl.model.PropertyMapping;
+import org.hibernate.search.elasticsearch.schema.impl.model.TypeMapping;
+import org.hibernate.search.util.impl.CollectionHelper;
+import org.hibernate.search.util.logging.impl.LoggerFactory;
+
+import com.google.gson.JsonPrimitive;
+
+/**
+ * The default {@link ElasticsearchSchemaValidator} implementation.
+ * <strong>Important implementation note:</strong> unexpected attributes (i.e. those not mapped to a field in TypeMapping)
+ * are totally ignored. This allows users to leverage Elasticsearch features that are not supported in
+ * Hibernate Search, by setting those attributes manually.
+ * @author Yoann Rodiere
+ */
+public class DefaultElasticsearchSchemaValidator implements ElasticsearchSchemaValidator {
+
+	private static final Log LOG = LoggerFactory.make( Log.class );
+
+	private static final double DEFAULT_DOUBLE_DELTA = 0.001;
+	private static final float DEFAULT_FLOAT_DELTA = 0.001f;
+
+	private static final List<String> DEFAULT_DATE_FORMAT;
+	static {
+		List<String> formats = new ArrayList<>();
+		formats.add( "strict_date_optional_time" );
+		formats.add( "epoch_millis" );
+		DEFAULT_DATE_FORMAT = CollectionHelper.toImmutableList( formats );
+	}
+
+	public DefaultElasticsearchSchemaValidator() {
+	}
+
+	@Override
+	public void validate(IndexMetadata expectedIndexMetadata, IndexMetadata actualIndexMetadata) {
+		// Unknown mappings are ignored, we only validate expected mappings
+		for ( Map.Entry<String, TypeMapping> entry : expectedIndexMetadata.getMappings().entrySet() ) {
+			String mappingName = entry.getKey();
+			TypeMapping expectedTypeMapping = entry.getValue();
+			TypeMapping actualTypeMapping = actualIndexMetadata.getMappings().get( mappingName );
+
+			if ( actualTypeMapping == null ) {
+				throw LOG.mappingMissing( mappingName );
```

Comment:
It is a bit strange that we abort on the first encountered mismatch.

Also, I find it strange that we throw an exception only to catch it in the client method.

Wouldn't it be nicer to collect all validation issues in one list, then throw a single clear exception with an explanation?